lib: Remove timeval-related build warnings

TAGOliveira201000 opened the pull request:
The futimes and lutimes functions were defined with a `struct timeval [2]` argument but they were instantiated with a pointer.

<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [ ] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): N/A
 - Platform(s): N/A
 - Application(s): any application using Musl


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

razvand commented:
@TAGOliveira201000 , please also fix the checkpatch warning above. Fix the commit.

TAGOliveira201000 committed:
lib/vfscore: Remove timeval-related build warnings

The futimes and lutimes functions are defined with a `struct timeval [2]`
argument. They are however instantiated with a pointer. This causes a build
warning.

Change the instantiation from pointer to array. This removes the build warning.

Signed-off-by: Tiago Andre Goncalves Oliveira <t.g.oliveira@ua.pt>

StefanJum commented:
Change `lib:` to `lib/vfscore:` in the commit message. All good besides that.

StefanJum reviewed (approved):
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
